### PR TITLE
[Build Output] Cleanup some well known messages and add tooltips to TreeView

### PR DESCRIPTION
**Pull request by rodrmoya**

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs [hunk 1]
```diff
@@ -142,24 +143,36 @@ private void BinLog_TaskFinished (object sender, TaskFinishedEventArgs e)
...
 		{
 			if (String.IsNullOrEmpty (e.Message)) {
 				return;
 			}
 
 			switch (e.Message[0]) {
+			case 'S':
+				if (e.Message.StartsWith (SkippingTargetMessagePrefix, StringComparison.Ordinal)) {
+					// "Skipping target ..." messages
+					var parts = e.Message.Split ('"');
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs]: Don't allocate here. Make a scanner, you can also do substring equality without allocating a substring.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs]: Same in below cases.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs [hunk 2]
```diff
@@ -142,24 +143,36 @@ private void BinLog_TaskFinished (object sender, TaskFinishedEventArgs e)
 
 			EndCurrentNode (stringPool.Add (e.Message), e.Timestamp);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs]: Several lines above, still adding the same string to the stringpool twice.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs (hunk 1 above)
Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs]: "Make a scanner", you mean use .IndexOf + .Compare, which don't do allocations?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs (hunk 2 above)
Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/MSBuildOutputProcessor.cs]: yes, this is being fixed in #3880 so not changing it here

**Pull request merged**